### PR TITLE
[improvement](scanner) push down late arrival runtime filters into SegmentIterator

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

File: be/src/olap/predicate_creator.h [hunk 1]
```diff
@@ -203,36 +257,76 @@ std::unique_ptr<PredicateCreator<ConditionType>> get_creator(const FieldType& ty
...
-        return std::make_unique<CustomPredicateCreator<TYPE_DATETIMEV2, PT, ConditionType>>(
-                timestamp_from_datetime_v2);
+        if constexpr (std::is_same_v<std::string, ConditionType> ||
+                      std::is_same_v<std::vector<std::string>, ConditionType>) {
+            return std::make_unique<CustomPredicateCreator<TYPE_DATETIMEV2, PT, ConditionType>>(
+                    timestamp_from_datetime_v2);
+        } else {
+            return std::make_unique<CustomPredicateCreator<TYPE_DATETIMEV2, PT, ConditionType>>(
+                    [](const void* const& condition) {
+                        return *reinterpret_cast<const uint32_t*>(condition);
```
Comment (yiguolei) [inline be/src/olap/predicate_creator.h:303]: uint32 ？？？

File: be/src/olap/reader.cpp [hunk 2]
```diff
@@ -237,6 +239,7 @@ Status TabletReader::_capture_rs_readers(const ReaderParams& read_params) {
     _reader_context.read_orderby_key_columns =
             _orderby_key_columns.size() > 0 ? &_orderby_key_columns : nullptr;
     _reader_context.predicates = &_col_predicates;
+    _reader_context.late_arrival_predicates = &_late_arrival_predicates;
```
Comment (yiguolei) [inline be/src/olap/reader.cpp:242]: This method will be called at the beginning of the scanner, If new predicates arrived, is it will affect??

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run p0 20

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run p0 10

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buidlall

**Comment by mrhhsg** [comment]
run buildall

**Pull request closed without merging**